stellargraph/layer/rgcn.py

```diff
@@ -0,0 +1,496 @@
+    """
+
+    def __init__(
+        self,
+        layer_sizes,
+        generator,
+        bias=True,
+        num_bases=0,
+        dropout=0.0,
+        activations=None,
+        **kwargs
+    ):
+        if not isinstance(generator, RelationalFullBatchNodeGenerator):
+            raise TypeError(
+                "Generator should be a instance of RelationalFullBatchNodeGenerator"
+            )
+
+        n_layers = len(layer_sizes)
+        self.layer_sizes = layer_sizes
+        self.activations = activations
+        self.bias = bias
+        self.num_bases = num_bases
+        self.dropout = dropout
+        self.generator = generator
+        self.support = 1
+
+        # Check if the generator is producing a sparse matrix
+        self.use_sparse = generator.use_sparse
+
+        # Activation function for each layer
+        if activations is None:
+            activations = ["relu"] * n_layers
+
+        elif len(activations) != n_layers:
+            raise ValueError(
+                "Invalid number of activations; require one function per layer"
+            )
+
+        self.activations = activations
+        self.num_bases = num_bases
+
+        # Optional regulariser, etc. for weights and biases
+        self._get_regularisers_from_keywords(kwargs)
+
+        # Initialize a stack of RGCN layers
+        self._layers = []
+        for ii in range(n_layers):
+            self._layers.append(Dropout(self.dropout))
+            self._layers.append(
+                RelationalGraphConvolution(
+                    self.layer_sizes[ii],
+                    num_relationships=len(generator.As),
+                    num_bases=self.num_bases,
+                    activation=self.activations[ii],
+                    use_bias=self.bias,
+                    final_layer=ii == (n_layers - 1),
+                    **self._regularisers
+                )
+            )
+
+    def _get_regularisers_from_keywords(self, kwargs):
+        regularisers = {}
+        for param_name in [
+            "kernel_initializer",
+            "kernel_regularizer",
+            "kernel_constraint",
+            "bias_initializer",
+            "bias_regularizer",
+            "bias_constraint",
+        ]:
+            param_value = kwargs.pop(param_name, None)
+            if param_value is not None:
+                regularisers[param_name] = param_value
+        self._regularisers = regularisers
+
+    def __call__(self, x):
+        """
+        Apply a stack of RGCN layers to the inputs.
+        The input tensors are expected to be a list of the following:
+        [Node features shape (1, N, F), Output indices (1, O)] +
+        [Adjacency indices for each relationship (1, E, 2) for _ in range(R)]
+        [Adjacency values for each relationshiop (1, E) for _ in range(R)]
+
+
+        where N is the number of nodes, F the number of input features,
+              E is the number of edges, O the number of output nodes,
```

Comment:
Can you please use another symbol instead of O? When I first saw it, I thought it was a zero. Maybe use Z?